tests/test_requests.py

```diff
@@ -606,6 +621,115 @@ def test_unicode_header_name(self, httpbin):
     def test_pyopenssl_redirect(self, httpbin_secure, httpbin_ca_bundle):
         requests.get(httpbin_secure('status', '301'), verify=httpbin_ca_bundle)
 
+    def test_verify(self):
+        """verify=True works with requests.get"""
+        with pytest.raises(SSLError):
+            requests.get('https://testssl-expire-r2i2.disig.sk/index.en.html')
+
+        with pytest.raises(SSLError):
+            requests.get('https://testssl-expire-r2i2.disig.sk/index.en.html')
+
+        with pytest.raises(SSLError):
+            requests.get('https://self-signed.badssl.com/')
+
+        with pytest.raises(SSLError):
+            requests.get('https://self-signed.badssl.com/')
+
+        session = requests.Session()
+
+        with pytest.raises(SSLError):
+            session.get('https://testssl-expire-r2i2.disig.sk/index.en.html')
+
+        with pytest.raises(SSLError):
+            session.get('https://testssl-expire-r2i2.disig.sk/index.en.html')
+
+        with pytest.raises(SSLError):
+            session.get('https://self-signed.badssl.com/')
+
+        with pytest.raises(SSLError):
+            session.get('https://self-signed.badssl.com/')
```

Comment:
I don't believe there's any need to have it be re-useable. urllib3 already has its socket-level tests which are well equipped for this kind of testing, and you could use requests' own socket-level tests to reproduce this function as well.